tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -202,3 +217,224 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
       assert.include hls, anchor12.highlight[anchor12.startPage]
       assert.include hls, anchor2.highlight[anchor2.startPage]
       assert.notInclude hls, anchor3.highlight[anchor3.startPage]
+
+  describe 'two-phased anchoring', ->
+
+    class DummyDocumentAccess
+
+      @applicable: -> true
+
+      isPageMapped: (index) -> index in @_rendered
+
+      # Helper function to trigger a page rendering
+      renderPage: (index) ->
+        return if @isPageMapped index
+
+        @_rendered.push index
+
+        # Publish an event
+        event = document.createEvent "UIEvents"
+        event.initUIEvent "docPageMapped", false, false, window, 0
+        event.pageIndex = index
+        window.dispatchEvent event
+
+      # Helper function to trigger a page unrendering
+      unrenderPage: (index) ->
+        return unless @isPageMapped index
+
+        i = @_rendered.indexOf index
+        @_rendered.splice(i, 1)
+
+        # Publish an event
+        event = document.createEvent "UIEvents"
+        event.initUIEvent "docPageUnmapped", false, false, window, 0
+        event.pageIndex = index
+        window.dispatchEvent event
```

Comment:
On 2015-02-10 10:46, Aron Carroll wrote:
> These helper functions are not part of the |DummyDocumentAccess| object 

In fact they kind of are, in the sense that normally, it's the
responsibility of the document access policy to monitor the platform
(which is doing the actual rendering), and to notify Annotator (or, to
be more concrete, the anchoring subsystem) about any page rendering events.

So these events are generally emitted by the document access policy;
that's why I put it there.

(But technically, I could just as well as put them into a separate
function, if you would prefer that.)